hack/push-updates.sh

```diff
@@ -0,0 +1,161 @@
+EOF
+
+    for r in $DEFAULT_REPOS; do
+        echo "      $r"
+    done
+
+cat <<EOF
+
+  -s SHA
+    The SHA from machine-api-operator repository to push.
+    Defaults to HEAD of master.
+
+  -t TITLE
+    The commit and pull-request title to use. Defaults to
+    a message that includes the SHA.
+
+Without any arguments, creates a pull request for each repository to
+update the vendored version of machine-api-operator to refer to the
+current HEAD commit on the master branch.
+
+EOF
+}
+
+DEFAULT_REPOS="
+cluster-api-provider-aws
+cluster-api-provider-azure
+cluster-api-provider-baremetal
+cluster-api-provider-gcp
+cluster-api-provider-openstack
+cluster-api-provider-ovirt
+"
+TITLE=""
+SHA=""
+BRANCH_NAME="master"
+DRY_RUN=false
+
+while getopts "b:hnr:s:t:" OPTION; do
+    case $OPTION in
+        b)
+            BRANCH_NAME="$OPTARG"
+            ;;
+        h)
+            show_help
+            exit 0
+            ;;
+        n)
+            DRY_RUN=true
+            ;;
+        r)
+            REPOS="$REPOS $OPTARG"
+            ;;
+        s)
+            SHA="$OPTARG"
+            ;;
+        t)
+            TITLE="$OPTARG"
+            ;;
+    esac
+done
+
+if ! which hub 2>/dev/null 1>&2; then
+    echo "This tool requires the hub command line app." 1>&2
+    echo "Refer to https://github.com/github/hub for installation instructions." 1>&2
+    exit 1
+fi
+
+REPOS=${REPOS:-${DEFAULT_REPOS}}
+TMPDIR=${TMPDIR:-/tmp}
+WORKING_DIR=$(mktemp -d $TMPDIR/mao-push-updates.XXXX)
+cd $WORKING_DIR
+echo "Working in $WORKING_DIR"
+
+if [ -z "$SHA" ]; then
+    echo "Determining SHA for machine-api-operator repository..."
+    hub clone openshift/machine-api-operator
+    FULL_SHA=$(cd ./machine-api-operator && git show --format="%H" origin/$BRANCH_NAME)
+    SHA=$(cd ./machine-api-operator && git show --format="%h" origin/$BRANCH_NAME)
+fi
+echo "Updating consumers to $SHA"
+
+if [ -z "$TITLE" ]; then
+    TITLE="Update machine-api-operator to ${SHA}"
+fi
+
+WORKING_BRANCH_NAME="mao-update-$SHA"
+REMOTE_NAME="working"
+
+set -e
+
+function fork_repo () {
+    local repo_name="$1"
+    local remote_name="$2"
+
+    if ! hub fork --remote-name $REMOTE_NAME; then
+        # Forking failed, so maybe we have another repo with that name
+        # forked from a different source. Try adding that as a remote.
+        echo "Forking failed, trying a simple clone"
+        local github_user=$(git config --get github.user)
+        git remote add $REMOTE_NAME git@github.com:${github_user}/${repo_name}.git
+        git remote update $REMOTE_NAME
+    fi
+}
+
+for repo in $REPOS; do
+    echo
+    echo $repo
+
+    hub clone openshift/$repo
+    pushd ./$repo
+
+    git checkout origin/$BRANCH_NAME
+    git checkout -b $WORKING_BRANCH_NAME
+
+    set -x
+
+    # Use -d option to avoid build errors because we aren't going to
+    # build locally.
+    go get -d github.com/openshift/machine-api-operator@${FULL_SHA}
```

Comment:
I don't think FULL_SHA is initialised when SHA is set explicitly, looks like it's only set when SHA is empty and we work it out in the script?

Reply:
I fixed that, and also split the operation into 2 loops as we discussed in the meeting this morning.